plantdescriptionengine/src/main/java/eu/arrowhead/core/plantdescriptionengine/alarms/AlarmManager.java

```diff
@@ -0,0 +1,263 @@
+        final Map<String, String> metadata
+    ) {
+        Objects.requireNonNull(systemId, "Expected system ID");
+        raiseAlarm(systemId, systemName, metadata, AlarmCause.MULTIPLE_MATCHES);
+    }
+
+    /**
+     * Raise an alarm indicating that a supposedly monitorable system does not
+     * seem to provide a 'Monitorable' service.
+     *
+     * @param systemId   ID of a Plant Description system.
+     * @param systemName Name of a Plant Description system, or null.
+     * @param metadata   Metadata of a Plant Description system, or null.
+     */
+    public void raiseSystemNotMonitorable(
+        final String systemId,
+        final String systemName,
+        final Map<String, String> metadata
+    ) {
+        Objects.requireNonNull(systemId, "Expected system ID");
+        raiseAlarm(systemId, systemName, metadata, AlarmCause.NOT_MONITORABLE);
+    }
+
+    /**
+     * Clear any alarms indicating that the system matching the given arguments
+     * does not provide a 'Monitorable' service.
+     *
+     * @param systemId   ID of a Plant Description system.
+     * @param systemName Name of a Plant Description system, or null.
+     * @param metadata   Metadata of a Plant Description system, or null.
+     */
+    public void clearSystemNotMonitorable(
+        final String systemId,
+        final String systemName,
+        final Map<String, String> metadata
+    ) {
+        Objects.requireNonNull(systemId, "Expected system ID");
+        final List<Alarm> cleared = activeAlarms.stream()
+            .filter(alarm ->
+                alarm.getCause() == AlarmCause.NOT_MONITORABLE &&
+                    Objects.equals(systemId, alarm.getSystemId()) &&
+                    Objects.equals(systemName, alarm.getSystemName()) &&
+                    Objects.equals(metadata, alarm.getMetadata())
+            )
+            .collect(Collectors.toList());
+
+        for (final Alarm alarm : cleared) {
+            clearAlarm(alarm);
+        }
+    }
+
+    /**
+     * Raise an alarm indicating that a monitorable system did not respond to
+     * a ping request.
+     *
+     * @param systemName Name of a Plant Description system.
+     */
+    public void raiseNoPingResponse(final String systemName) {
+        Objects.requireNonNull(systemName, "Expected system name.");
+        raiseAlarm(null, systemName, Collections.emptyMap(), AlarmCause.NO_PING_RESPONSE);
+    }
+
+    /**
+     * Clear any alarms indicating that the given system did not respond to a
+     * ping request.
+     *
+     * @param systemName Name of a Plant Description system.
+     */
+    public void clearNoPingResponse(final String systemName) {
+        Objects.requireNonNull(systemName, "Expected system name.");
+
+        final List<Alarm> cleared = activeAlarms.stream()
+            .filter(alarm ->
+                alarm.getCause() == AlarmCause.NO_PING_RESPONSE &&
+                    systemName.equals(alarm.getSystemName())
```

Comment:
maybe system name normalization can cause issues

Reply:
System names are now validated at the time they enter the PDE. This means that the system names used here are trimmed and on the correct format.